modin/experimental/xgboost/xgboost.py

```diff
@@ -135,6 +336,31 @@ def predict(
             data, DMatrix
         ), f"Type of `data` is {type(data)}, but expected {DMatrix}."
 
+        if (
+            self.feature_names is not None
+            and data.feature_names is not None
+            and self.feature_names != data.feature_names
+        ):
+            dat_missing = set(self.feature_names) - set(data.feature_names)
+            my_missing = set(data.feature_names) - set(self.feature_names)
+
+            msg = "feature_names mismatch: {0} {1}"
+
+            if dat_missing:
+                msg += (
+                    "\nexpected "
+                    + ", ".join(str(s) for s in dat_missing)
+                    + " in input data"
+                )
+
+            if my_missing:
+                msg += (
+                    "\ntraining data did not have the following fields: "
+                    + ", ".join(str(s) for s in my_missing)
+                )
+
+            raise ValueError(msg.format(self.feature_names, data.feature_names))
+
```

Comment:
What's need of changes in `Booster`?

Reply:
This need because in this lines raise error which check in `test_feature_names`